presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -346,12 +345,13 @@ private static int hashBytes(int initialValue, Slice bytes)
         return Optional.of(new HiveBucketFilter(builder.build()));
     }
 
-    private static Optional<Set<Integer>> getHiveBuckets(Table table, Map<ColumnHandle, Set<NullableValue>> bindings)
+    private static Optional<Set<Integer>> getHiveBuckets(Optional<HiveBucketProperty> hiveBucketPropertyOptional, List<Column> dataColumns, Map<ColumnHandle, Set<NullableValue>> bindings)
```

Comment:
- `hiveBucketProperty`
- Keep the original signature `private static Optional<Set<Integer>> getHiveBuckets(Table table, Map<ColumnHandle, Set<NullableValue>> bindings)` and create a new one so that the old one will use the new one. This will avoid the change to `HivePartitionManager`